libs/src/utils/network.js

```diff
@@ -62,10 +62,40 @@ async function timeRequestsAndSortByVersion (requests, timeout = null) {
     // If health check failed, send to back of timings
     if (!a.response) return 1
     if (!b.response) return -1
+
     // Sort by highest version
     if (semver.gt(a.response.data.data.version, b.response.data.data.version)) return -1
     if (semver.lt(a.response.data.data.version, b.response.data.data.version)) return 1
-    // If same version, do a tie breaker on the response time
+
+    // Sort by content node transcode queue load
+    // defined as the ratio of (active + waiting transcodes) / (transcode slots, same as number of cpu cores)
+    // if either number of CPUs or transcode fields aren't defined in the health check, ratios default to 1
+    let numCPUsA = 1
+    let numCPUsB = 1
+    if (a.response.data.data && a.response.data.data.numberOfCPUs) {
+      numCPUsA = a.response.data.data.numberOfCPUs
+    }
+    if (b.response.data.data && b.response.data.data.numberOfCPUs) {
+      numCPUsB = b.response.data.data.numberOfCPUs
+    }
+
+    let transcodeCountA = numCPUsA
+    let transcodeCountB = numCPUsB
+    if (a.response.data.data && a.response.data.data.hasOwnProperty('transcodeActive') && a.response.data.data.hasOwnProperty('transcodeWaiting')) {
+      transcodeCountA = a.response.data.data.transcodeActive + a.response.data.data.transcodeWaiting
+    }
+    if (b.response.data.data && b.response.data.data.hasOwnProperty('transcodeActive') && b.response.data.data.hasOwnProperty('transcodeWaiting')) {
+      transcodeCountB = b.response.data.data.transcodeActive + b.response.data.data.transcodeWaiting
+    }
+
+    // lower is better, means less transcode queue utilization
+    // default ratio for transcode count and num cpus if node doesn't have one of those defined is 1
```

Comment:
i think this leads to unintended behavior if one or both nodes haven't defined transcode counts
CN1 = 8 numCPUs
CN2 = 4 numCPUs -> 20 transcodeCount
outcome: 8/8 = 1 & 20/4 = 5; 1 < 5; CN1 preferred. that seems incorrect? what if CN1 has 50 transcodeCount

feels like we should only consume transcodeCounts if both nodes provide them, else only use numCPUs which we know all nodes provide

Reply:
my rationale was if one reports transcode count and the other doesnt, it's likely behind in version so the higher version will get placed ahead anyway. this case should only really get hit if both nodes don't have something